mlos_bench/mlos_bench/optimizers/base_optimizer.py

```diff
@@ -317,27 +307,33 @@ def _get_score(self, status: Status, score: Optional[Union[float, Dict[str, floa
         ----------
         status : Status
             Final status of the experiment (e.g., SUCCEEDED or FAILED).
-        score : Union[float, Dict[str, float]]
-            A scalar or a dict with the final benchmark results.
+        score : Optional[Dict[str, TunableValue]]
+            A dict with the final benchmark results.
             None if the experiment was not successful.
 
         Returns
         -------
-        score : float
-            A scalar benchmark score to be used as a primary target for MINIMIZATION.
+        score : Optional[Dict[str, float]]
+            An optional dict of benchmark scores to be used as targets for MINIMIZATION.
         """
         if not status.is_completed():
             return None
-        if status.is_succeeded():
-            assert score is not None
-            if isinstance(score, dict):
-                if self._opt_target not in score:
-                    raise ValueError(f"Missing expected optimization target metric '{self._opt_target}' "
-                                     + f"in results for iteration {self._iter}: {score}")
-                score = score[self._opt_target]
-            return float(score) * self._opt_sign
-        assert score is None
-        return float("inf")
+
+        if not status.is_succeeded():
+            assert score is None
+            return {
+                opt_target: float("inf")
```

Comment:
does this need to be adjusted based on target direction?

Reply:
it does not - that's an internal function and it assumes that `self._opt_targets` has already been adjusted for target direction.

Reply:
But, down below we do the multiplication by the opt_direction and here we don't.

Doesn't that mean that for unsuccessful trials, we always report `float("inf")` even if the `opt_dir` is `"max"` for that `opt_target`?

That seems like a bug to me.  Do we have a test case checking for this?

Reply:
`Optimizer._get_scores()` _always_ returns scores adjusted for **minimization** (I mention that in the docstring). That's why we do the multiplication here. If the trial is not successful, for minimization we must always return _positive_ infinity (or later maybe some other large _positive_ value).

I've added a unit test to verify that behavior